spec/requests/events_request_spec.rb

```diff
@@ -0,0 +1,194 @@
+          expect(response).to redirect_to business_room_path(id: valid_params[:room_id],
+            day: valid_params[:date_event].strftime(Settings.format_year_month_date))
+        end
+
+        it "flash success" do
+          expect(flash[:success]).to match I18n.t("controller.events.success_create")
+        end
+      end
+
+      context "with invalid attributes" do
+        before {post :create, params: {event: invalid_params}}
+
+        it "create a new event fail" do
+          expect{subject}.to change(Event, :count).by(0)
+        end
+
+        it "render template room" do
+          expect(response).to render_template "business/rooms/show"
+        end
+
+        it "create a new event fail" do
+          expect(flash[:error]).to match I18n.t("controller.events.error_create")
+        end
+      end
+    end
+
+    describe "GET #edit" do
+      context "with valid event" do
+        before{get :edit, params: {id: event.id}}
+        it "find event & render view edit template" do
+          expect(response).to render_template "business/events/edit"
+        end
+      end
+      context "with invalid product" do
+        before{get :edit, params: {id: "asdasd"}}
+        it "not find event & redirect to shops_products_path" do
+          expect(response).to redirect_to business_home_path
+        end
+      end
+    end
+
+    describe "PATCH #update" do
+      before {
+        patch :update, params: {
+          id: event.id,
+          event: {
+            title: "test",
+            date_event: "2020-09-03",
+            room_id: event.room_id
+          }
+        }
+      }
+      context "with valid attributes" do
+        it "flash success" do
+          expect(flash[:success]).to match(I18n.t("controller.events.success_update"))
+        end
+
+        it "redirect business_room_path" do
+          expect(response).to redirect_to business_room_path(id: event.room_id, day: "2020-09-03")
+        end
+      end
+      context "with invalid attributes" do
+        before {
+          patch :update, params: {
+            id: event.id,
+            event: {
+              date_event: "2020-09-06",
+              room_id: event.room_id
+            }
+          }
+        }
+        it "update error" do
+          expect(flash[:error]).to match I18n.t("controller.events.error_update")
+        end
+
+        it "render template" do
+          expect(response).to redirect_to business_room_path event.room_id
+        end
+      end
+    end
+
+    describe "DELETE #destroy" do
+      context "with valid product" do
+        let!(:event_destroy) {FactoryBot.create :event, user_id: user.id, room_id: room.id, date_event: "2020/09/04"}
+        before{delete :destroy, params: {id: event.id}}
+
+        it "delete success" do
+          expect{
+          delete :destroy, params: {id: event_destroy.id}
+            }.to change(Event, :count).by(-1)
+        end
+
+        it " show flash message" do
```

Comment:
```suggestion
        it "show flash message" do
```
k check lại trước khi ready hả